qa/tasks/cephfs/test_misc.py

```diff
@@ -524,6 +524,34 @@ def test_session_ls(self):
     def test_client_ls(self):
         self._session_client_ls(['client', 'ls'])
         
+    
+class TestSessionClientEvict(CephFSTestCase):
+    CLIENTS_REQUIRED = 3
+
+    def _session_client_evict(self, cmd):
+        mount_a_client_id = self.mount_a.get_global_id()
+        info = self.fs.rank_asok(cmd + ['ls'])
+        self.assertEqual(len(info), 3)
+        with self.assertRaises(CommandFailedError) as ce:
+            self.fs.rank_tell(cmd + ['evict'])
+        self.assertEqual(ce.exception.exitstatus, errno.EPERM)
+        info = self.fs.rank_asok(cmd + ['ls'])
+        self.assertEqual(len(info), 3)
+        self.fs.rank_asok(cmd + ['evict', f'id={mount_a_client_id}'])
+        info = self.fs.rank_asok(cmd + ['ls'])
+        self.assertEqual(len(info), 2)
+        self.assertNotIn(mount_a_client_id, [val['id'] for val in info])
+        self.fs.rank_asok(cmd + ['evict', '--yes-i-really-mean-it'])
+        info = self.fs.rank_asok(cmd + ['ls'])
+        self.assertEqual(len(info), 0) # multiple clients are evicted
```

Comment:
Suggest breaking these into separate tests. Please also check:

* behavior of `--help` flag (implies empty filter which should be caught)
* behavior of non-existent flag (like `--help`) but something we'll never implement in the future like `--foo`
* behavior with `id=0`
* behavior with `id=*`
* behavior with non-existent id like `id=1`
* behavior with `id=<negative integer>`

Reply:
Added these checks

Reply:
It's a tad bit better to have separate tests covering the scenarios @batrick mentions.

Reply:
I did split the tests